entities/src/main/java/org/odk/collect/entities/javarosa/intance/LocalEntitiesInstanceAdapter.kt

```diff
@@ -38,9 +38,9 @@ class LocalEntitiesInstanceAdapter(private val entitiesRepository: EntitiesRepos
         }
     }
 
-    fun queryEq(instanceId: String, child: String, value: String): List<TreeElement>? {
-        return when {
-            child == EntityItemElement.ID -> {
+    fun queryEq(instanceId: String, child: String, value: String): List<TreeElement> {
```

Comment:
Now as this function can't return null you can simplify `LocalEntitiesFilterStrategy#filter`.